lib/logstash/inputs/kafka.rb

```diff
@@ -98,7 +104,7 @@ def register
       options[:reset_beginning] = 'from-beginning'
     end # if :reset_beginning
     @kafka_client_queue = SizedQueue.new(@queue_size)
-    @consumer_group = Kafka::Group.new(options)
+    @consumer_group = @kafka_group_class.new(options)
```

Comment:
@jsvd I was giving a thought on what you said, and makes not much of sense to me. 

* I agree with dependency is not going to change over time, at least for now, and that we saw this change because of an issue with RSpec3.
* But if we want to do what you just said, we will be just somewhere with a mid way solution, (removing the dependency injection) would not really change the whole approach, and the default parameter is basically doing the job.
* As I can see form https://github.com/joekiller/jruby-kafka/blob/master/lib/jruby-kafka/group.rb#L1, we this looks like being a kind of consumer, so at the end we might end up with using different kind of consumers, so this might be a good design change.

What do you think?